src/main/java/net/imglib2/algorithm/hough/HoughLineTransform.java

```diff
@@ -0,0 +1,292 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package net.imglib2.algorithm.hough;
+
+import java.util.ArrayList;
+
+import net.imglib2.Cursor;
+import net.imglib2.img.Img;
+import net.imglib2.img.ImgFactory;
+import net.imglib2.type.Type;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.type.NativeType;
+import net.imglib2.type.numeric.integer.IntType;
+import net.imglib2.type.numeric.integer.LongType;
+import net.imglib2.type.numeric.integer.ShortType;
+
+/**
+ * A class that extends {@link HoughTransform} to handle Hough Line voting over an edge map.
+ * This implementation uses a threshold to determine whether a pixel at a certain point is
+ * an edge or not.  Comparison is strictly-greater-than.  This implementation is fairly dumb
+ * in that it does not take gradients into account.  The threshold used is the default value
+ * returned by calling the constructor for the {@link Type} of the input {@link Img}.
+ *
+ * Vote space here has two dimensions: rho and theta.  Theta is measured in radians
+ * [-pi/2 pi/2), rho is measured in [-rhoMax, rhoMax).
+ *
+ * Lines are modeled as
+ *
+ * l(t) = | x | = rho * |  cos(theta) | + t * | sin(theta) |
+ *        | y |         | -sin(theta) |       | cos(theta) |
+ *
+ * In other words, rho represents the signed minimum distance from the image origin to the line,
+ * and theta indicates the angle between the row-axis and the minimum offset vector.
+ *
+ * For a given point, then, votes are placed along the curve
+ *
+ * rho = y * sin(theta) + x * cos(theta)
+ * @Override
+ *
+ */
+public class HoughLineTransform<S extends RealType<S> & NativeType<S>, T extends Type<T> & Comparable<T>>
+        extends HoughTransform<S, T> {
+    public static final int DEFAULT_THETA = 180;
+    public final double dTheta;
+    public final double dRho;
+    private final T threshold;
+    private final int nRho;
+    private final int nTheta;
+    private final double[] rho;
+    private final double[] theta;
+    private ArrayList<double[]> rtPeaks;
+
+    final private static float computeLength(final long[] position) {
```

Comment:
Please apply the ImgLib2 [style formatting](https://github.com/imglib/imglib2/tree/master/doc) to this class, so formatting is consistent with the rest of ImgLib2.